### ISSUE TITLE
PyBDSF doesn't build with pybuilder

### ISSUE THREAD

**Issue by gijzelaerr** [lofar-astron/PyBDSF#45]
outlog log
https://gist.github.com/gijzelaerr/644c4d6cde70e5ef026b6e84fa151c03

pybuilder is a useful tool for creating Debian packages. Probably it doesn't build since it copies all relevant files in a subfolder. Probably some file is missing. A normal `build_ext` seems to work. 
I don't see a MANIFEST.in file, probably that is the problem.

**Comment by gijzelaerr** [comment]
MANIFEST.in file should probably look something like this:
```
include natgrid/Include/*.h 
include natgrid/Src/*.c 
include src/minpack/*.f
include src/port3/*.f      
include src/fortran/*.f    
include src/fortran/constants.inc
include src/c++/num_util/*.cpp   
include src/c++/num_util/*.h       
include src/c++/*.h                      
include src/c++/*.cc
```
but that doesn't seem to solve the problem. Maybe I'm missing a file.

**Comment by o-smirnov** [comment]
Can somebody take a look at this please? @tammojan?

**Comment by tammojan** [comment]
The first error I see in the [log](https://gist.github.com/gijzelaerr/644c4d6cde70e5ef026b6e84fa151c03) is
```
/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
```
Perhaps [this stackexchange post](https://stackoverflow.com/questions/11116399/crt1-o-in-function-start-undefined-reference-to-main-in-linux) is relevant?

It surprises me that the arguments for `gfortran` on [line 162](https://gist.github.com/gijzelaerr/644c4d6cde70e5ef026b6e84fa151c03#file-buildlog-L162) has no `-shared` option, while it is generating a `.so` file:
```
/usr/bin/gfortran -Wall -g -Wl,-Bsymbolic-functions -Wl,-z,relro build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/bdsf/_pytesselatemodule.o build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/fortranobject.o build/temp.linux-x86_64-2.7/src/fortran/pytess_simple.o build/temp.linux-x86_64-2.7/src/fortran/pytess_roundness.o -lgfortran -o /vault-tina/gijs/packaging/build/pybdsf/.pybuild/pythonX.Y_2.7/build/bdsf/_pytesselate.so
```

The undefined references (`PyNumber_Int`, etc.) are all in the python C api, to which I also see no references in the above call that fails.

**Comment by o-smirnov** [comment]
Yeah @gijzelaerr it looks like the remaining problem is in the build scripts. Probably a missing ``-shared`` flag, so it tries to build the .so as an executable, hence the missing main, hence the missing Python symbols.

**Comment by gijzelaerr** [comment]
this patch fixes the problem https://github.com/kernsuite-debian/pybdsf/blob/master/debian/patches/setup.patch

**Comment by tammojan** [comment]
A patch for this was merged in #51
